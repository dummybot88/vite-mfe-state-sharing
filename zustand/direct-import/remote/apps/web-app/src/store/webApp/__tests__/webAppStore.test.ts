import { cleanup, renderHook } from '@testing-library/react'
import useWebAppStore, { setIssues } from '../webAppStore'
import { act } from 'react'
describe('store/webApp/webAppStore', () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let result: { current: any }
  beforeEach(() => {
    result = renderHook(() => useWebAppStore()).result
  })
  afterEach(cleanup)

  it('should fetch initial webapp state', () => {
    // Assert
    expect(result.current).toMatchSnapshot()
  })

  it('should update the store with the issues for the webapp.', () => {
    expect(result.current.issues).toEqual([])

    // Act
    act(() =>
      setIssues([
        {
          isbn: '1234567890',
          abstract: 'This is an abstract'
        }
      ])
    )

    // Assert
    expect(result.current.issues).toEqual([
      {
        isbn: '1234567890',
        abstract: 'This is an abstract'
      }
    ])
  })
})
