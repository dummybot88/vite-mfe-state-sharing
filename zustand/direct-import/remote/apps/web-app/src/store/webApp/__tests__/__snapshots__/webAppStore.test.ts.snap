// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store/webApp/webAppStore should fetch initial webapp state 1`] = `
{
  "issues": [],
}
`;
